wagtail_transfer/auth.py

```diff
@@ -1,9 +1,37 @@
 import hashlib
 import hmac
+import re
 
 from django.conf import settings
 from django.core.exceptions import PermissionDenied
 
+GROUP_QUERY_WITH_DIGEST = re.compile('(?P<query_before>.*?)&?digest=(?P<digest>[^&]*)(?P<query_after>.*)')
+
+def check_get_digest_wrapper(view_func):
+    """
+    Check the digest of a request matches its GET parameters
+    This is useful when wrapping vendored API views
+    """
+    def decorated_view(request, *args, **kwargs):
+        query_string = request.META.get('QUERY_STRING', '')
+        match = GROUP_QUERY_WITH_DIGEST.match(query_string)
+        if not match:
+            raise PermissionDenied
+        digest = match.group('digest')
+        message = f'{match.group("query_before")}{match.group("query_after")}'
+        if not (digest and message):
+            # This decorator is intended for use with GET parameters
+            # If there are none, or no digest, something's gone wrong
+            raise PermissionDenied
+        check_digest(message, digest)
+
+        # Unfortunately the admin API views won't allow unknown GET parameters
+        # So we must remove the digest parameter from the request as well
+        request.META['QUERY_STRING'] = message
```

Comment:
Slightly surprised that this and the old code that updated request.GET both work - I would have expected that the request.GET dict would be constructed before this point, and that DRF would look at that. I guess it must be lazily evaluated on first access instead. TIL :-)

Reply:
I was surprised too, tbh! Thinking of this though, I've added an extra commit to remove any GET that is cached on the request, just in case someone evaluates it in middleware or some other way before this decorator gets to rewrite the query string

Reply:
Good plan! Looks good to me.